devicetypes/qubino/qubino-3-phase-meter.src/qubino-3-phase-meter.groovy

```diff
@@ -0,0 +1,194 @@
+			}
+		}
+		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat",width: 2, height: 2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		standardTile("configure", "device.power", inactiveLabel: false, decoration: "flat",width: 2, height: 2) {
+			state "configure", label:'', action:"configuration.configure", icon:"st.secondary.configure"
+		}
+
+		main (["power","energy"])
+		details(["power","energy","refresh", "configure"])
+	}
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	state.numberOfMeters = 3
+
+	if (!childDevices) {
+		addChildMeters(state.numberOfMeters)
+	}
+
+	response(refresh())
+}
+
+def updated() {
+	response(refresh())
+}
+
+def ping() {
+	refresh()
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = createEvent(zwaveEvent(cmd))
+	}
+	log.debug "Parse returned ${result?.descriptionText}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(versions)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, ep = null) {
+	log.debug "Multichannel command ${cmd}" + (ep ? " from endpoint $ep" : "")
+	if (cmd.commandClass == 0x6C && cmd.parameter.size >= 4) { // Supervision encapsulated Message
+		// Supervision header is 4 bytes long, two bytes dropped here are the latter two bytes of the supervision header
+		cmd.parameter = cmd.parameter.drop(2)
+		// Updated Command Class/Command now with the remaining bytes
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, endpoint = null) {
+	handleMeterReport(cmd, endpoint)
+}
+
+private handleMeterReport(cmd, endpoint) {
+	def event = createMeterEventMap(cmd)
+	if (endpoint && endpoint > 1) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(event)
+	} else {
+		createEvent(event)
+	}
+}
+
+private createMeterEventMap(cmd) {
+	def eventMap = [:]
+	if (cmd.scale == 0) {
+		eventMap.name = "energy"
+		eventMap.value = cmd.scaledMeterValue
+		eventMap.unit = "kWh"
+	} else if (cmd.scale == 2) {
+		eventMap.name = "power"
+		eventMap.value = cmd.scaledMeterValue >= 0 ? Math.round(cmd.scaledMeterValue) : Math.round(cmd.scaledMeterValue) * (-1)
```

Comment:
Were you getting negative power reports? Seems weird.

But if that is expected, you could probably just do `Math.round(Math.abs())`

Reply:
Actually this is expected 
![image](https://user-images.githubusercontent.com/41617389/92711498-e0a80200-f358-11ea-942e-b542ffc9b898.png)
But it sometimes sends positive values anyway...
And that I haven't thought about _abs_ is, well, embarasing ;_;